test/tests/restart/restart_subapp_not_master/two_step_solve_sub_restart.i

```diff
@@ -0,0 +1,82 @@
+[Mesh]
+  file = two_step_solve_master_cp/0002_mesh.cpr
+[]
+
+[Problem]
+  restart_file_base = two_step_solve_master_cp/LATEST
+  allow_sub_app_restart = true
```

Comment:
Ok - sounds good
On Wed, Mar 9, 2016 at 2:31 PM Cody Permann <notifications@github.com>
wrote:

> In
> test/tests/restart/restart_subapp_not_master/two_step_solve_sub_restart.i
> <https://github.com/idaholab/moose/pull/6531#discussion_r55574630>:
>
> > @@ -0,0 +1,82 @@
> > +[Mesh]
> > +  file = two_step_solve_master_cp/0002_mesh.cpr
> > +[]
> > +
> > +[Problem]
> > +  restart_file_base = two_step_solve_master_cp/LATEST
> > +  allow_sub_app_restart = true
>
> Confirmed. It appears that all subapps have the MooseApp::_restart flag
> set if the master is restarting. There's currently no way to differentiate
> between the master restarting and having a subapp restarting in FEProblem.
> I can see why this flag needs to be set but we need to be able to tell them
> apart. Let's get this merged so that the Rattlesnake guys can continue
> working on the sprint problem. We can deal with making this cleaner when we
> work on #5877 <https://github.com/idaholab/moose/issues/5877>. That may
> be needed sooner than later, but not today!
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/idaholab/moose/pull/6531/files#r55574630>.
>